### DISCUSSION TITLE
Get different column from relationship for calculations?

### DISCUSSION THREAD

**Discussion by Seriousness** [Budibase/budibase#5094] [Help]
I have tried to find the right wording but could not really find a similar discussion (whilst there pretty surely might be one :D)

I do have a relationship to a meta-table named "months", so for selection I have a costs-table where I can select Months by names. For calculations I need the according number-column of these months.

so eg. my row displays like:

Title: Cleaning
Months: 
   Jan[id=1]
   March[id=3]

for calculations I now need the according id of the assigned Months in that costs-row. Is that possible with handlebars|js? If yes, how? I do understand that there might be better ways and in that case no metatable might be needed, but I have several other ideas that rely on that concept, so this is more a tryout.

Hope my question is understandible ;)

**Comment by melohagan** [comment]
The default BudibaseDB is a NoSQL CouchDB instance, and has no query options for creating views of joined data. 

If I understand the question correctly, the best way to do relational views is with a relational database. 

For example you could add a **MySQL** datasource. Then click on 'Add query'.

Now you can write a JOIN and transformer to get the data you need from both tables:

![Screenshot 2022-03-28 at 15 56 35](https://user-images.githubusercontent.com/101575380/160426505-83560f84-61ed-490b-91a7-036d13ed0f6a.png)

Then you would have a table block that uses the query as its datasource:

![Screenshot 2022-03-28 at 15 57 50](https://user-images.githubusercontent.com/101575380/160426777-0325a3c1-870a-4d42-bdc0-b164982a8c3f.png)

Let me know if that helps!
